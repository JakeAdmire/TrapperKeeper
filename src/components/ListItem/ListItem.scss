ul.ListItems {
  margin: 0;
  padding: 0;

  button.CheckBox {
    transition: all .5s ease;

    &:hover {
      opacity: .7;
    }
  }

  .ItemIncomplete:last-child {





  }


  li.ItemIncomplete {
    position: relative;
    transition: all .3 ease;

    &:last-child {

      .CheckBox {
        img { display: none; }
        position: relative;
        pointer-events: none;
        background: url('./media/plus.svg') no-repeat;
        left: 13.5px;
        top: 4px;
        opacity: .5;
      }

      input.list-item-input {
        &::placeholder {}
      }

      .delete-button {
        opacity: 0;
        pointer-events: none;
      }
    }

    &:not(:last-child) {

      .delete-button {
        opacity: 0;
        pointer-events: none;
        transition: all .3 ease;
      }

      &:hover {
        .delete-button {
          opacity: .6;
          pointer-events: all;
          &:hover {
            opacity: .9;
          }
        }
      }

      &:focus-within {

        .delete-button {
          opacity: .6;
          pointer-events: all;
        }

        &:before {
          content: '';
          position: absolute;
          height: 1px;
          width:100%;
          top: 0px;
          border-top: 1px solid rgba(0, 0, 0, 0.15);
        }
        &:after {
          content: '';
          position: absolute;
          height: 1px;
          width:100%;
          bottom: 0px;
          border-bottom: 1px solid rgba(0, 0, 0, 0.15);
        }

      }

      input.list-item-input {
        text-decoration: none;

        &::placeholder {
          color: transparent;
        }
      }

      .CheckBox:hover + .list-item-input {
        text-decoration: line-through;
      }

    }

  }

  li.ItemComplete {

    input.list-item-input {
      text-decoration: line-through;
    }

    .CheckBox:hover + .list-item-input {
      text-decoration: none;
    }
  }

  li {
    list-style-type: none;
    display: flex;
    justify-content: space-between;
    padding: 3px 0;

    .list-item-input {
      font-family: $font-Roboto;
      font-size: .9rem;
      font-weight: 400;
      line-height: 1.5rem;
    }

    label {
      width: 4%;
    }

    input[type=text] {
      outline: none;
      border:none;
      width: 80%;
    }

    button {
      background: none;
      outline: none;
      width: 10%;
      border: none;
      cursor: pointer;
      line-height: 1;
    }

    img {
      height: 18px;
      opacity: .54;
    }
  }
}
