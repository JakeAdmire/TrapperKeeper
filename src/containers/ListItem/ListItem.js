// import React from 'react';

// export const ListItem = (props) => {

//   const { id, index, isComplete, text, handleItemChange, handleItemDelete, toggleComplete } = props;

//   return (
//     <li key={index}>
//       <label className="container">
//         <input onClick={() => toggleComplete(id)} type="checkbox" className={isComplete ? 'checkbox checked' : 'checkbox not-checked'}/>
//         <span className="checkmark"></span>
//       </label>
//       <input type="text"
//               placeholder="List Item"
//               value={text}
//               name="ListItem"
//               className="list-item-input"
//               onChange={(e) => handleItemChange(e, id, index)}
//               disabled={isComplete ? 'disabled' : null}
//         />
//         <button className="delete-button" onClick={(e) => handleItemDelete(e, index)}>
//           <i className="fas fa-times"></i>
//         </button>
//     </li>
//   )

// }

import React, { Component } from 'react';

export class ListItem extends Component {
  
  render() {
    const { id, index, isComplete, text, handleItemChange, handleItemDelete, toggleComplete } = this.props;
    return (
      <li key={index}>
        <label className="container">
          <input onClick={() => toggleComplete(id)} type="checkbox" className={isComplete ? 'checkbox checked' : 'checkbox not-checked'}/>
          <span className="checkmark"></span>
        </label>
        <input type="text"
                placeholder="List Item"
                value={text}
                name="ListItem"
                className="list-item-input"
                onChange={(e) => handleItemChange(e, id, index)}
                disabled={isComplete ? 'disabled' : null}
                id={id}
          />
          <button className="delete-button" onClick={(e) => handleItemDelete(e, index)}>
            <i className="fas fa-times"></i>
          </button>
      </li>
    )
  }
}