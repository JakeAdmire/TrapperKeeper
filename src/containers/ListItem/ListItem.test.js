import React from 'react';
import { shallow } from 'enzyme';
import { ListItem } from './ListItem';


describe('ListItem', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ListItem />)
  })

  it('should match the snapshot', () => {

  })

})
