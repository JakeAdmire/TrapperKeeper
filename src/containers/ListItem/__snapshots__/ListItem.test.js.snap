// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match the snapshot 1`] = `
<li
  key="0"
>
  <label
    className="container"
  >
    <input
      className="checkbox not-checked"
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
  <input
    className="list-item-input"
    name="ListItem"
    onChange={[Function]}
    placeholder="List Item"
    type="text"
  />
  <button
    className="delete-button"
    onClick={[Function]}
  >
    <i
      className="fas fa-times"
    />
  </button>
</li>
`;
